test_conformance/math_brute_force/common.cpp

```diff
@@ -530,3 +552,50 @@ __kernel void )", kernel_name.c_str(), R"((__global RETTYPE_SCALAR* out,
 
     return kernel.str();
 }
+
+cl_int BuildKernels(BuildKernelInfo &info, cl_uint job_id,
+                    SourceGenerator generator)
+{
+    // Generate the kernel code.
+    cl_uint vector_size_index = gMinVectorSizeIndex + job_id;
+    auto kernel_name = GetKernelName(vector_size_index);
+    auto source = generator(kernel_name, info.nameInCode, vector_size_index);
+    std::array<const char *, 1> sources{ source.c_str() };
+
+    // Create the program.
+    clProgramWrapper &program = info.programs[vector_size_index];
+    auto options = GetBuildOptions(info.relaxedMode);
+    int error =
+        create_single_kernel_helper(gContext, &program, nullptr, sources.size(),
+                                    sources.data(), nullptr, options.c_str());
+    if (error != CL_SUCCESS)
+    {
+        vlog_error("\t\tFAILED -- Failed to create program. (%d)\n", error);
+        return error;
+    }
+
+    // Create a kernel for each thread. cl_kernels aren't thread safe, so make
+    // one for every thread
+    auto &kernels = info.kernels[vector_size_index];
+    assert(kernels.empty() && "Dirty BuildKernelInfo");
+    kernels.resize(info.threadCount);
+    for (auto &kernel : kernels)
+    {
+        kernel = clCreateKernel(program, kernel_name.c_str(), &error);
+        if (!kernel || error != CL_SUCCESS)
+        {
+            vlog_error("\t\tFAILED -- clCreateKernel() failed: (%d)\n", error);
+            size_t log_size;
+            clGetProgramBuildInfo(program, gDevice, CL_PROGRAM_BUILD_LOG, 0,
+                                  nullptr, &log_size);
+            std::string buffer;
+            buffer.resize(log_size + 1);
+            clGetProgramBuildInfo(program, gDevice, CL_PROGRAM_BUILD_LOG,
+                                  log_size, &buffer[0], NULL);
```

Comment:
Definitely not a must-fix since this was in the original code as well, but on our implementation at least the build log would only be useful when the program failed to build and wouldn't be useful when clCreateKernel failed.

Reply:
Good observation; I suppose we can remove the `clGetProgramBuildInfo` queries here, as the spec says this will "Return the build, compile or link log for `clBuildProgram`, `clCompileProgram` or `clLinkProgram`", i.e., nothing about `clCreateKernel`.

I'll add it to my list of future cleanups.